dev/Interop/CameraCaptureUI/CameraCaptureUI/CameraCaptureUIPhotoCaptureSettings.cpp

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+#include "pch.h"
+#include "CameraCaptureUIPhotoCaptureSettings.h"
+#include "Microsoft.Windows.Media.Capture.CameraCaptureUIPhotoCaptureSettings.g.cpp"
+
+using namespace winrt::Windows::Foundation;
+
+namespace winrt::Microsoft::Windows::Media::Capture::implementation
+{
+    void CameraCaptureUIPhotoCaptureSettings::validate()
+    {
+        Size const size = CroppedSizeInPixels;
+        Size const aspect = CroppedAspectRatio;
+        auto hasPhotoFixedSizeConstraint = (size.Width != 0) || (size.Height != 0);
+        auto hasPhotoAspectRatioConstraint = (aspect.Width != 0) || (aspect.Height != 0);
+        auto hasPhotoSizeConstraint = hasPhotoFixedSizeConstraint || hasPhotoAspectRatioConstraint;
+
+        if (hasPhotoAspectRatioConstraint && hasPhotoFixedSizeConstraint)
+        {
+            throw hresult_invalid_argument(L"PhotoSettings can't have both a size and aspect ratio specified");
+        }
+        else if (!AllowCropping && hasPhotoSizeConstraint)
+        {
+            throw hresult_invalid_argument(L"PhotoSettings can't have a ratio or size specified with cropping disabled");
+        }
+    }
+
+    hstring CameraCaptureUIPhotoCaptureSettings::GetFileExtension()
+    {
+        switch (Format)
+        {
+        case CameraCaptureUIPhotoFormat::Jpeg:
+            return L".jpg";
```

Comment:
The enum value is `jpeg` but the returned string is `jpg`.  Should these be consistent?  (As best as I can tell both are valid exceptions for a JPEG image).

Reply:
That is how things are in the original API, kept it the same to maintain parity.